samples/client/petstore/typescript-angular-v4.3/npm/api/pet.service.ts

```diff
@@ -156,8 +171,13 @@ export class PetService {
      * Finds Pets by status
      * Multiple status values can be provided with comma separated strings
      * @param status Status values that need to be considered for filter
+     * @param observe set whether or not to return the data Observable as the body, response or events. defaults to returning the body.
+     * @param reportProgress flag to report request and response progress.
      */
-    public findPetsByStatus(status: Array<string>): Observable<Array<Pet>> {
+    public findPetsByStatus(status: Array<string>, observe?: 'body', reportProgress?: boolean): Observable<Array<Pet>>;
+    public findPetsByStatus(status: Array<string>, observe?: 'response', reportProgress?: boolean): Observable<HttpResponse<Array<Pet>>>;
+    public findPetsByStatus(status: Array<string>, observe?: 'events', reportProgress?: boolean): Observable<HttpEvent<Array<Pet>>>;
+    public findPetsByStatus(status: Array<string>, observe: any = 'body', reportProgress: boolean = false ): Observable<any> {
```

Comment:
The type of observe seems too permissive here, because it would allow invocations like:

```
findPetsByStatus(status, observe: HttpResponse)
findPetsByStatus(status, observe: "Body")
```

Angular's HttpClient restricts this parameter to permitted values:
```
observe?: HttpObserve,
```
where
```
/**
 * @stable
 */
export type HttpObserve = 'body' | 'events' | 'response';
```

So we should be able to import that type, and use it ourselves? Is there a reason we can't follow angular's lead here?

Reply:
you can't import it.  I tried it myself and it did not work.  You can try it in your project if you'd like and tell me if i'm doing something wrong.

Reply:
also it doens't allow those invocations.  you are only allowed to pass thoes specific things cause it's the string type.

Reply:
the last observe: any is not counted for the invocations.  only the overloads before it.

Reply:
@bedag-moo from angular public_api.d.ts
export { HttpClient } from './src/client';

they don't export HttpObserve

Reply:
My apologies, I didn't know that typescript treats does not consider the last signature part of the overload list. Your code is fine then, sorry for not checking.